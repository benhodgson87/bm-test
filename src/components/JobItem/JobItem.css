.job-item {
  display: flex;
  align-items: center;
  padding: 10px;
  border: 1px solid transparent;
  border-left: 5px solid #2673b4;
  background-color: #fff;

  &:hover {
    border-width: 1px 1px 1px 5px;
    border-color: #2673b4;
    background-color: #e9f2f8;
  }
}

.job-item:not(:last-of-type) {
  margin-bottom: 4px;
}
