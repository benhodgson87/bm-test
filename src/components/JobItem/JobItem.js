import React from 'react'
import PropTypes from 'prop-types'

const JobItem = ({ name, dimensions }) => (
  <div>
    {name} - {dimensions}sqft
  </div>
)

JobItem.propTypes = {
  name: PropTypes.string.isRequired,
  dimensions: PropTypes.number.isRequired,
}

export default JobItem
