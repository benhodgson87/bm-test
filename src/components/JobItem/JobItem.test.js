import React from 'react'
import { shallow } from 'enzyme'

import JobItem from './JobItem'

const baseProps = {
  id: 123,
  customer: {
    title: 'Mr',
    first_name: 'Wayne',
    last_name: 'World',
  },
  dimensions: 234,
}

const renderProps = props => Object.assign({}, baseProps, props)

describe('JobItem component', () => {
  test('renders as expected', () => {
    const wrapper = shallow(<JobItem {...renderProps({})} />)
    expect(wrapper).toMatchSnapshot()
  })
})
