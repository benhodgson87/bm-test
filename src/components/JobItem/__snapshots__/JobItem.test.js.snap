// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem component renders as expected 1`] = `
<Row
  className="job-item"
>
  <Col
    className="job-item-section"
    xs={5}
  >
    Mr
     
    Wayne
     
    World
  </Col>
  <Col
    className="job-item-section"
    xs={3}
  >
    <strong>
      234
    </strong>
    ftÂ³
  </Col>
  <Col
    className="job-item-section"
    xs={2}
  >
    <Label
      bsClass="label"
      bsStyle="primary"
    >
      Cube Job
    </Label>
  </Col>
  <Col
    className="job-item-section"
    xs={2}
  >
    <Connect(JobActionsContainer)
      id={123}
    />
  </Col>
</Row>
`;
